foundation-ui/auth/AuthService.js

```diff
@@ -0,0 +1,67 @@
+import {EventEmitter} from 'events';
+
+import Authorizer from './Authorizer';
+import {CHANGE} from './AuthEvent';
+
+/**
+ * List of authorizers
+ *
+ * @type {Array.<Authorizer>}
+ */
+const authorizers = [];
+
+/**
+ * AuthService
+ */
+class AuthService extends EventEmitter {
+
+  /**
+   * Register authorizer
+   *
+   * @param {Authorizer} authorizer
+   */
+  registerAuthorizer(authorizer) {
+    if (!authorizer || !(authorizer instanceof Authorizer)) {
+      if (global.__DEV__) {
+        throw new TypeError('Provided component must be a ' +
+            'React.Component constructor or a stateless functional component.');
+      }
+
+      return;
+    }
+
+    authorizers.push(authorizer);
+    this.emit(CHANGE);
+  }
+
+  /**
+   * Unregisters authorizer
+   *
+   * @param  {Authorizer} authorizer
+   */
+  unregisterAuthorizer(authorizer) {
+    let i = authorizers.length;
+    while (--i >= 0) {
+      if (authorizers[i] === authorizer) {
+        authorizers.splice(i, 1);
+        break;
+      }
+    }
+
+    this.emit(CHANGE);
+  }
+
+  /**
+   * Authorize current user if required permission was granted
+   *
+   * @param  {string} permission
+   *
+   * @returns {boolean} authorized
+   */
+  authorize(permission = '') {
+    return authorizers.every((authorizer) => authorizer.authorize(permission));
+  }
+
+}
+
+module.exports = new AuthService();
```

Comment:
You are right, this singleton pattern makes it rather hard to test the service properly. I thought about introducing a new pattern but decided to follow our current "standard" way for consistency.

I started a document back in May illustrating a few alternative patterns and shared it with @dcos/frontend and would love your feedback as we finally need to agree on a new pattern due to the mentioned reasons.